core/cmd/remote_client.go

```diff
@@ -623,7 +647,12 @@ func parseResponse(resp *http.Response) ([]byte, error) {
 	if resp.StatusCode == http.StatusUnauthorized {
 		return b, errUnauthorized
 	} else if resp.StatusCode >= http.StatusBadRequest {
-		return b, errors.New("Error")
+		errorMessage, err := parseErrorResponseBody(b)
+
+		if err != nil {
+			return b, errors.New("Error")
```

Comment:
Here you swallow `err` and we never know what was it.
Let's at least return that `err` instead of your custom error, or wrap that err with a custom message (see the examples in the code around).